### PR TITLE
add 'metrics/slis' to kcm health checks

### LINKED ISSUE #112958: Wire up `metrics/slis` for kube-controller-manager
### What would you like to be added?

As per, https://kep.k8s.io/3466, wire up `metrics/slis` for KCM

### Why is this needed?

https://kep.k8s.io/3466

### PR DESCRIPTION
**Pull request by logicalhan**
Change-Id: I8c2114e538bb417deff8c3f9f107758c089227dc

#### What type of PR is this?

/kind feature

Fixes #112958 

#### What this PR does / why we need it:

Adds health check SLI metrics to KCM. This is behind a feature gate. 

#### Does this PR introduce a user-facing change?

```release-note
If `ComponentSLIs` feature gate is enabled, then `/metrics/slis` becomes available on kube-controller-manager, allowing you to scrape health check metrics.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://kep.k8s.io/3466
```

```
# HELP kubernetes_healthcheck [ALPHA] This metric records the result of a single healthcheck.
# TYPE kubernetes_healthcheck gauge
kubernetes_healthcheck{name="attachdetach",type="healthz"} 1
kubernetes_healthcheck{name="clusterrole-aggregation",type="healthz"} 1
kubernetes_healthcheck{name="cronjob",type="healthz"} 1
kubernetes_healthcheck{name="csrapproving",type="healthz"} 1
kubernetes_healthcheck{name="csrcleaner",type="healthz"} 1
kubernetes_healthcheck{name="csrsigning",type="healthz"} 1
kubernetes_healthcheck{name="daemonset",type="healthz"} 1
kubernetes_healthcheck{name="deployment",type="healthz"} 1
kubernetes_healthcheck{name="disruption",type="healthz"} 1
kubernetes_healthcheck{name="endpoint",type="healthz"} 1
kubernetes_healthcheck{name="endpointslice",type="healthz"} 1
kubernetes_healthcheck{name="endpointslicemirroring",type="healthz"} 1
kubernetes_healthcheck{name="ephemeral-volume",type="healthz"} 1
kubernetes_healthcheck{name="garbagecollector",type="healthz"} 1
kubernetes_healthcheck{name="horizontalpodautoscaling",type="healthz"} 1
kubernetes_healthcheck{name="job",type="healthz"} 1
kubernetes_healthcheck{name="namespace",type="healthz"} 1
kubernetes_healthcheck{name="nodelifecycle",type="healthz"} 1
kubernetes_healthcheck{name="persistentvolume-binder",type="healthz"} 1
kubernetes_healthcheck{name="persistentvolume-expander",type="healthz"} 1
kubernetes_healthcheck{name="podgc",type="healthz"} 1
kubernetes_healthcheck{name="pv-protection",type="healthz"} 1
kubernetes_healthcheck{name="pvc-protection",type="healthz"} 1
kubernetes_healthcheck{name="replicaset",type="healthz"} 1
kubernetes_healthcheck{name="replicationcontroller",type="healthz"} 1
kubernetes_healthcheck{name="resourcequota",type="healthz"} 1
kubernetes_healthcheck{name="root-ca-cert-publisher",type="healthz"} 1
kubernetes_healthcheck{name="serviceaccount",type="healthz"} 1
kubernetes_healthcheck{name="statefulset",type="healthz"} 1
kubernetes_healthcheck{name="ttl",type="healthz"} 1
kubernetes_healthcheck{name="ttl-after-finished",type="healthz"} 1
# HELP kubernetes_healthchecks_total [ALPHA] This metric records the results of all healthcheck.
# TYPE kubernetes_healthchecks_total counter
kubernetes_healthchecks_total{name="attachdetach",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="clusterrole-aggregation",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="cronjob",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="csrapproving",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="csrcleaner",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="csrsigning",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="daemonset",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="deployment",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="disruption",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="endpoint",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="endpointslice",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="endpointslicemirroring",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="ephemeral-volume",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="garbagecollector",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="horizontalpodautoscaling",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="job",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="namespace",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="nodelifecycle",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="persistentvolume-binder",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="persistentvolume-expander",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="podgc",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="pv-protection",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="pvc-protection",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="replicaset",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="replicationcontroller",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="resourcequota",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="root-ca-cert-publisher",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="serviceaccount",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="statefulset",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="ttl",status="success",type="healthz"} 2
kubernetes_healthchecks_total{name="ttl-after-finished",status="success",type="healthz"} 2
```

### CODE REVIEW

**Review by logicalhan — COMMENTED** [review]
/sig cloud-provider
/assign @cheftako

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -205,6 +208,9 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 	var unsecuredMux *mux.PathRecorderMux
 	if c.SecureServing != nil {
 		unsecuredMux = genericcontrollermanager.NewBaseHandler(&c.ComponentConfig.Generic.Debugging, healthzHandler)
+		if utilfeature.DefaultFeatureGate.Enabled(features.ComponentSLIs) {
```
[the remark this answers is not in the document]
Comment (logicalhan) [reply cmd/kube-controller-manager/app/controllermanager.go:211]: I'm confused... why?

[the remark this answers is not in the document]
Comment (logicalhan) [reply cmd/kube-controller-manager/app/controllermanager.go:211]: filed https://github.com/kubernetes/kubernetes/issues/112982

**Comment by cheftako** [comment]
Do we have a way to point users at the meaning for values 1 or 2?

**Pull request merged**